sql/src/main/java/org/apache/druid/sql/calcite/schema/SystemSchema.java

```diff
@@ -313,7 +316,10 @@ public Enumerable<Object[]> scan(DataContext root)
                   (long) segment.getShardSpec().getPartitionNum(),
                   numReplicas,
                   numRows,
-                  IS_PUBLISHED_TRUE, //is_published is true for published segments
+                  //is_active is true for published segments that are not overshadowed
+                  val.isOvershadowed() ? IS_ACTIVE_FALSE : IS_ACTIVE_TRUE,
```

Comment:
mmm.. isn't it a requirement for being active that is_overshadow and is_publish both be true?

Reply:
Yes, that's the idea. The branch is for published segments only.